hyperspy/api.py

```diff
@@ -21,10 +21,10 @@
 import logging
 
 import hyperspy.api_nogui
-from hyperspy.api_nogui import __doc__, get_configuration_directory_path
-from hyperspy.defaults_parser import preferences
-from hyperspy.logger import set_log_level
-from . import __version__
+from hyperspy.api_nogui import __doc__, get_configuration_directory_path # noqa: F401
+from hyperspy.defaults_parser import preferences # noqa: F401
```

Comment:
There seems to be lots of unused imports here and in the api_nogui files.  Maybe some comment as to why that is the case?

Reply:
@ericpre Sounds good to me!  I always was suspicious that this was the case but never tracked to logic to see.  

I'm in favor of merging this and then figuring that out in a separate PR.